core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -599,8 +609,15 @@ sealed trait ZIO[-R, +E, +A] extends Serializable with ZIOPlatformSpecific[R, E,
   final def collectZIO[R1 <: R, E1 >: E, B](e: => E1)(pf: PartialFunction[A, ZIO[R1, E1, B]]): ZIO[R1, E1, B] =
     self.flatMap(v => pf.applyOrElse[A, ZIO[R1, E1, B]](v, _ => ZIO.fail(e)))
 
+  @deprecated("the Arrow combinators are deprecated", "2.0.0")
   final def compose[R1, E1 >: E](that: ZIO[R1, E1, R]): ZIO[R1, E1, A] = self <<< that
 
+  /**
+   * Transforms the environment.
+   */
+  final def contramap[R0](f: R0 => R)(implicit ev: NeedsEnv[R]): ZIO[R0, E, A] =
```

Comment:
Not sure there is a lot of value in changing this if we are going to leave `provide`. If we want to take back the `provide` name and use it for new layer operators I think that is interesting but I think otherwise we're just fragmenting things splitting up the old layer operators between `provide` and `contramap` versus `provide` and `provideSome`.